Первая половина проекта

AniraUoren opened the pull request:
Здесь сделана первая половина проекта на React.

AniraUoren committed:
feat: AppHeader

Собран заголовок

Closes WP-226 Step 1 - AppHeader

AniraUoren committed:
feat: AppHeader

Поправлены стили заголовка

Closes WP-226 Step 1 - AppHeader

AniraUoren committed:
feat: BurgerIngredients

Сделан блок ингредиентов

Closes WP-227 Step 1 - BurgerIngredients

AniraUoren committed:
feat: Стили

Сделан блок ингредиентов

AniraUoren committed:
feat: Стили для заголовка

1. Переделана разметка: Конструктор, Лента заказов и Личный кабинет стали ссылками.
2. Попали в PP.

AniraUoren committed:
fix: Поправлен импорт

AniraUoren committed:
fix: Поправлены стили до PP

AniraUoren committed:
fix: Поправлены стили до PP

AniraUoren committed:
fix: Поправлены стили до PP

убраны outline

AniraUoren committed:
fix: Поправлена разметка

gennady-bars commented on src/components/burger-constructor/burger-constructor.jsx:
```diff
@@ -0,0 +1,115 @@
+import Styles from "./burger-constructor.module.css";
+
+function BurgerConstructor() {
+    const [cart, setCart] = React.useState([
+        {
+            "_id": "60666c42cc7b410027a1a9b1",
+            "name": "Краторная булка N-200i",
+            "type": "bun",
+            "price": 1255,
+            "image": "https://code.s3.yandex.net/react/code/bun-02.png"
+        },
+        {
+            "_id": "60666c42cc7b410027a1a9b9",
+            "name": "Соус традиционный галактический",
+            "type": "sauce",
+            "price": 15,
+            "image": "https://code.s3.yandex.net/react/code/sauce-03.png"
+        },
+        {
+            "_id": "60666c42cc7b410027a1a9b4",
+            "name": "Мясо бессмертных моллюсков Protostomia",
+            "type": "main",
+            "price": 1337,
+            "image": "https://code.s3.yandex.net/react/code/meat-02.png"
+        },
+        {
+            "_id": "60666c42cc7b410027a1a9bc",
```
Моковые данные лучше вынести в `utils` и оттуда импортировать, чтобы не засорять код компонентов

gennady-bars commented on src/components/burger-constructor/burger-constructor.jsx:
```diff
@@ -0,0 +1,115 @@
+                        extraClass="ml-8"
+                    />
+                </div>
+                <div style={{maxHeight: "568px",display: 'flex', flexDirection: 'column', gap: '16px', overflowY: "scroll"}}
```
Инлайн-стилей `style={{`не должно быть в проекте, если они не меняются **динамически** в коде компонента. Это затрудняет управление стилями. Все стили описываются в модульных `.module.css` файлах, а в коде компонентов используются модульные `css`-стили.

gennady-bars commented on src/components/burger-ingredients/burger-ingredients.jsx:
```diff
@@ -0,0 +1,77 @@
+                Соберите бургер
+            </p>
+            <div className="mb-10">
+                <div style={{display: 'flex'}} className="mb-10">
```
Инлайн-стилей `style={{`не должно быть в проекте, если они не меняются **динамически** в коде компонента. Это затрудняет управление стилями. Все стили описываются в модульных `.module.css` файлах, а в коде компонентов используются модульные `css`-стили.

gennady-bars commented on src/components/burger-ingredients/burger-ingredients.jsx:
```diff
@@ -0,0 +1,77 @@
+                        Начинки
+                    </Tab>
+                </div>
+                <div className="custom-scroll" style={{overflowY: "scroll", maxHeight: "756px", maxWidth: "600px"}}>
```
Инлайн-стилей `style={{`не должно быть в проекте, если они не меняются **динамически** в коде компонента. Это затрудняет управление стилями. Все стили описываются в модульных `.module.css` файлах, а в коде компонентов используются модульные `css`-стили.

gennady-bars commented on src/components/burger-ingredient/burger-ingredient.jsx:
```diff
@@ -0,0 +1,21 @@
+
+import {CurrencyIcon, Counter} from "@ya.praktikum/react-developer-burger-ui-components";
+
+function BurgerIngredient(props) {
```
`BurgerIngredient` забыли типизировать

gennady-bars replied:
Нужно посмотреть все компоненты и **типизировать** их `пропсы`, если где-то пропустили.

gennady-bars commented on src/components/burger-ingredients/burger-ingredients.jsx:
```diff
@@ -0,0 +1,77 @@
+                                data.map(elem => {
+                                    if (elem.type === "main") {
+                                        return (
+                                            <BurgerIngredient ingredient={elem} key={elem._id}/>
```
` key={elem._id}`

 Отлично, что используете атрибут `key`, и отлично, что для него используете уникальный `_id`, а не индекс массива.

[Почему индекс не нужно использовать, можно прочитать тут]( https://habr.com/ru/companies/hh/articles/352150/)

gennady-bars reviewed (commented):
Здравствуйте. (**Нужно развернуть общий комментарий ↓**)

Посмотрите на **гитхабе** все комментарии к коду (нужно прокрутить вниз страницу там)

Работа проделана огромная:



* Хорошая верстка
* Отлично, что не забываете про тег `main` для главной части страницы
* Отлично, что используете атрибут `key`, и отлично, что для него используете уникальный `_id`, а не индекс массива.



но есть некоторые недочеты:




* Моковые данные лучше вынести в `utils` и оттуда импортировать, чтобы не засорять код компонентов
* В заказе нужно указывать иконку булки именно https://disk.yandex.ru/i/_5xyyQaQmMoDdg 
* Инлайн-стилей `style={{`не должно быть в проекте, если они не меняются **динамически** в коде компонента. Это затрудняет управление стилями. Все стили описываются в модульных `.module.css` файлах, а в коде компонентов используются модульные `css`-стили.
* `BurgerIngredient` забыли типизировать 
* Нужно посмотреть все компоненты и **типизировать** их `пропсы`, если где-то пропустили.

AniraUoren committed:
fix: Исправление замечаний мидл-кодревью 1

1. Убраны пропущенные инлайновые стили.
2. Для компонента с пропсами описаны типы.

AniraUoren committed:
fix: Исправление замечаний мидл-кодревью 1

3. Отрисовывается правильная картинка для булочки.

gennady-bars commented on src/components/burger-ingredient/burger-ingredient.jsx:
```diff
@@ -18,4 +19,13 @@ function BurgerIngredient(props) {
 
+BurgerIngredient.propTypes = {
+    ingredient: PropTypes.shape({
+        image: PropTypes.string,
```
В ингредиенте очень много данных, а не только 3 шт.


Нужно описать весь тип в `prop-types.js` (у Вас есть этот файл уже)

gennady-bars replied:
Тип для ингредиента (propTypes) повторяется в некоторых компонентах. Не нужно его копировать. Это очень сложный объект. Нужно вынести его в константу `ingredientType` в отдельный файл `types.js` в папку `utils` и импортировать туда, где нужно

gennady-bars reviewed (commented):
1й этап принят. Нужно сделать теперь 2й с попапами. 

* В верхней булке нужно добавлять текст `верх`, а в нижней - `низ`. Посмотрите в фигме, пожалуйста
* Тип для ингредиента (propTypes) повторяется в некоторых компонентах. Не нужно его копировать. Это очень сложный объект. Нужно вынести его в константу `ingredientType` в отдельный файл `types.js` в папку `utils` и импортировать туда, где нужно
